[Site Design Revamp] Enlarges design thumbnails

antonis opened the pull request:
Fixes #16408

## Description
This PR enlarges the the thumbnails of the site design picker [by dynamically setting the layout dimension](https://github.com/wordpress-mobile/WordPress-Android/pull/16474/commits/3a3c9d53506bf46277b805f3bbd4126b272b7128). This is achieved by:
1. defining a [common dimension provider interface for both layout/design pickers](https://github.com/wordpress-mobile/WordPress-Android/pull/16474/commits/3a3c9d53506bf46277b805f3bbd4126b272b7128#diff-0af868e878e3142e191a89ea7d12c80bf5c38c188210b947e78346a50a0493dcR3)
2. passing a different implementation for[ the Page Layout picker
](https://github.com/wordpress-mobile/WordPress-Android/pull/16474/commits/3a3c9d53506bf46277b805f3bbd4126b272b7128#diff-cf1d16b94282b2fcde6bda1b642a0b1b585aa4e5f5beadbecf19bac0aa3f03b5R77) and [the Site Design picker](https://github.com/wordpress-mobile/WordPress-Android/pull/16474/commits/3a3c9d53506bf46277b805f3bbd4126b272b7128#diff-390502960224826eee61d0e68f9d1ffddfd5f7c9b63237b2d27f1b55ef7bfaceR66)
3. [setting the desired preview thumbnail dimensions programmatically](https://github.com/wordpress-mobile/WordPress-Android/pull/16474/commits/3a3c9d53506bf46277b805f3bbd4126b272b7128#diff-7cc5758d7a580ed0ef5cc0fe3e34ef38db330161a96af3a37356625e1ffa9e9eR38)

The PR also [sets the loading skeleton dimensions](https://github.com/wordpress-mobile/WordPress-Android/pull/16474/commits/ed9e0c6fa06346874a331085fdc480d17ba6b15b) to match those of the thumbnails on each screen.

Note:
Other alternative approaches were considered/tried. One involved creating separate layout files but due to the structure of the code this required duplicating the implementation of the `LayoutCategoryAdapter` and its nested components.

## To test
### Start the site creation flow and verify that the thumbnails are enlarged

|Before (`feature/site-design-revamp`)|After|
|---|---|
|![Screenshot_20220504_120733](https://user-images.githubusercontent.com/304044/166653235-39b4d7cf-632e-454c-a3e1-26c57beb282d.png)|![Screenshot_20220504_190123](https://user-images.githubusercontent.com/304044/166722666-86145004-522d-41d4-8ad7-13ba48770a70.png)|

### Verify that the UI of the page layout picker is unchanged

|Before (`19.8`)|After|
|---|---|
|![2Screenshot_20220504_120843](https://user-images.githubusercontent.com/304044/166652950-9900773a-55fb-4e7e-8e74-42ba1c4ba7ae.png)|![Screenshot_20220504_120827](https://user-images.githubusercontent.com/304044/166653001-e279feeb-4542-4e2a-a93b-a17ddeb48d9e.png)|

## Regression Notes
1. Potential unintended areas of impact
Page Layout Picker UI

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and the existing `ModalLayoutPickerViewModelTest`

3. What automated tests I added (or what prevented me from doing so)
Relied on existing tests

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

antonis committed:
Sets the layout picker thumbnail layout dimension dynamically

antonis committed:
Sets the loading skeleton layout dimension dynamically

twstokes commented:
👋 I've updated [the issue](https://github.com/wordpress-mobile/WordPress-Android/issues/16408) to include the exact dimensions from the designs, which it looks like you're matching for mobile (200px). 👍 

I'm not sure if Android will need to handle tablet widths, which appear to be 250px.

antonis committed:
Uses a different thumb dimension for tablet devices

antonis commented:
> 👋 I've updated [the issue](https://github.com/wordpress-mobile/WordPress-Android/issues/16408) to include the exact dimensions from the designs, which it looks like you're matching for mobile (200px). 👍
> 
> I'm not sure if Android will need to handle tablet widths, which appear to be 250px.

Thank you for pointing this out @twstokes 🙇 
I've added a different dimension for tablet with 3f080dea38ae988d110569c52920abf1780a0657

antonis committed:
Sets the category row height dynamically

mkevins reviewed (approved):
I tested this on a Pixel 3a (physical device) for both the page and site design pickers, in both portrait and landscape modes, and it is working as described. Also, the code changes look great.

Thank you for also mentioning the alternative approaches, and the rationale for going with this approach. I completely agree with the rationale, and appreciate the effort to find the cleanest way forward, reducing the potential for tensions to arise between the usages of these shared implementations. Great work Antonis!

One minor observation (not a blocker, but perhaps just something to confirm), I noticed that in the screenshots for the page picker provided in the description:

<details>
<summary>screenshots</summary>

### Verify that the UI of the page layout picker is unchanged

|Before (`19.8`)|After|
|---|---|
|![2Screenshot_20220504_120843](https://user-images.githubusercontent.com/304044/166652950-9900773a-55fb-4e7e-8e74-42ba1c4ba7ae.png)|![Screenshot_20220504_120827](https://user-images.githubusercontent.com/304044/166653001-e279feeb-4542-4e2a-a93b-a17ddeb48d9e.png)|
</details>

the first category (Blog) seems slightly smaller in the "after" image (i.e. the horizontal rules seem closer together vertically compared to the "before" image). I didn't notice this in my testing, but only when looking closely at the provided screenshots, so perhaps this is more of an issue with the screenshots themselves? I also notice the clock is offset a bit, so I'm thinking it's a screenshot issue, and not an implementation issue.

antonis commented:
Thank you for reviewing @mkevins 🙇 
> the first category (Blog) seems slightly smaller in the "after" image (i.e. the horizontal rules seem closer together vertically compared to the "before" image)
I hadn't noticed that but I think it has to do with the way the markdown table is rendered since when I open the full size images the match on the size of the rows.
![Screenshot 2022-05-05 at 12 38 16 PM](https://user-images.githubusercontent.com/304044/166898400-48632a2a-154b-4ecf-ab12-0349c55915ae.png)
I noticed thought the issue with the subtitle width/margin (notice that the 2nd line breaks on a different place) 🤔 
I'll proceed with merging this PR and look into it separately since this seem to exist on our feature branch already.

mkevins commented:
> when I open the full size images the match on the size of the rows

Thanks for confirming :+1:  I figured it must be from the way it was displayed, since the clock was not aligned either, and that's not part of "our" UI :smile: .
